arraykit.c

```diff
@@ -6,17 +6,40 @@
 
 # include "numpy/arrayobject.h"
 
+//------------------------------------------------------------------------------
+// Macros
+//------------------------------------------------------------------------------
+
 // Bug in NumPy < 1.16 (https://github.com/numpy/numpy/pull/12131):
 # undef PyDataType_ISBOOL
 # define PyDataType_ISBOOL(obj) \
     PyTypeNum_ISBOOL(((PyArray_Descr*)(obj))->type_num)
 
+// Given a PyObject, raise if not an array.
 # define AK_CHECK_NUMPY_ARRAY(O)                                              \
     if (!PyArray_Check(O)) {                                                  \
         return PyErr_Format(PyExc_TypeError, "expected numpy array (got %s)", \
                             Py_TYPE(O)->tp_name);                             \
     }
 
+// Given a PyObject, raise if not an array or is not one or two dimensional.
+# define AK_CHECK_NUMPY_ARRAY_1D_2D(O) \
+    AK_CHECK_NUMPY_ARRAY(O)\
+    int ndim = PyArray_NDIM((PyArrayObject *)O);\
+    if (ndim != 1 && ndim != 2) {\
+        return PyErr_Format(PyExc_NotImplementedError,\
+                "expected 1D or 2D array (got %i)",\
+                ndim);\
+    }
```

Comment:
Just a bit of cleanup. Here, the `do { ... } while (0)` idiom is even *more* important, since we don't want `ndim` to leak into the caller's scope!

```suggestion
    do {\
        AK_CHECK_NUMPY_ARRAY(O);\
        int ndim = PyArray_NDIM((PyArrayObject *)O);\
        if (ndim != 1 && ndim != 2) {\
            return PyErr_Format(PyExc_NotImplementedError,\
                                "expected 1D or 2D array (got %iD)", ndim);\
        }\
    } while (0)\
```

Reply:
Yes, I see. Thanks!